tests/cmdline/commands/test_data.py

```diff
@@ -332,6 +332,66 @@ def test_bandsexport(self):
         self.assertEqual(res.exit_code, 0, 'The command did not finish correctly')
         self.assertIn(b'[1.0, 3.0]', res.stdout_bytes, 'The string [1.0, 3.0] was not found in the bands' 'export')
 
+    def test_bandsexport_single_kp(self):
+        """ test issue #2462 """
+
+        # Create bands structure object.
+        alat = 4.  # angstrom
+        cell = [
+            [
+                alat,
+                0.,
+                0.,
+            ],
+            [
+                0.,
+                alat,
+                0.,
+            ],
+            [
+                0.,
+                0.,
+                alat,
+            ],
+        ]
+        strct = StructureData(cell=cell)
+        strct.append_atom(position=(0., 0., 0.), symbols='Fe')
+        strct.append_atom(position=(alat / 2., alat / 2., alat / 2.), symbols='O')
+        strct.store()
+
+        @calcfunction
+        def connect_structure_bands(strct):  # pylint: disable=unused-argument
```

Comment:
`create_structure_bands` is a static method, so perhaps you could simply reuse it?

Reply:
I need a single kpoint `BandsData` object. So I can not figure out how to reuse the cls function. The `bands` object has to be create as the offspring node of `StructureData`

Reply:
I think this is a compromise at the moment, it will be better to create an isolated `BandsData` after https://github.com/aiidateam/aiida-core/issues/3811 solved.

Reply:
I'll refactor this after https://github.com/aiidateam/aiida-core/pull/3817 is merged. @ltalirz but I don't whether should I directly change the code here, or update the branch to latest and then modified the testcase?